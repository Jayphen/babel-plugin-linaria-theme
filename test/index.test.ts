import main from "../src";
import pluginTester from "babel-plugin-tester";

pluginTester({
  plugin: main,
  snapshot: true,
  tests: [
    {
      title: "Array",
      pluginOptions: {
        minJSONStringSize: 0
      },
      code: `
const styles = css\`
  \${theme.below.sm} {
    color: red;
  };
  background: red;
  color: purple;
  margin: \${theme.default.sizes[1]};
  border: 1px solid \${theme.default.sizes[1]};
  border: 1px solid \${theme.default.sizes[1]};
  background: black;
  color: \${theme.default.colors.green};

      \``
    }
  ]
});
