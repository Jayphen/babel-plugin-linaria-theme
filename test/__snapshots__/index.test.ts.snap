// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-linaria-theme Array: Array 1`] = `

const styles = css\`
  \${theme.below.sm} {
    color: red;
  };
  background: red;
  color: purple;
  margin: \${theme.default.sizes[1]};
  border: 1px solid \${theme.default.sizes[1]};
  background: black;
  color: \${theme.default.colors.green};

      \`

      ↓ ↓ ↓ ↓ ↓ ↓

const styles = css\`
  \${theme.below.sm} {
    color: red;
  }
  background: red;
  color: purple;
  \${themed(t => ({
    margin: t.sizes[1]
  }))};
  \${themed(t => ({
    border: t.sizes[1]
  }))};
  background: black;
  \${themed(t => ({
    color: t.colors.green
  }))};
\`;


`;
